docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,359 @@
+
+The media type is defined as vnd.oci.image-compatibility.spec, in order to be defined agnostic to cluster management software.
+
+### Image Compatibility Spec
+
+The spec structure is quite simple, a flat json file.
+
+Schema:
+
+- **`compatibilities`** _list of compatibility sets_
+
+  This REQUIRED property describes all image compatibility sets, contains at least one compatibility set.
+
+  - **`compatibility set`** _list of compatibility requirements_
+
+    This REQUIRED property describes compatibility requirements in one compatibility set.
+
+    If one of the compatibility sets passes the compatibility check against the target node, that means the node is compatible with the image.
+
+    There are three category key-value pair inside the compatibility set.
+
+    - **`compatibility feature field`**
+
+      This REQUIRED property describes a feature compatibility requirement on the target host, the field accepts OCI and community defined labels.
+
+      The value of compatibility field in the compatibility artifact SHOULD be described in contiguous or discontinuous range if the value discovered on nodes can be different.
+
+      It's the tool's responsibility to interpret the value of OCI or community defined feature label, and matching criteria with the value discovered from the node via NFD tool or other tool. NFD means the Kubernetes node feature discovering SIG.
+
+      Inevitably, some compatibility fields MAY be duplicated among compatibility sets.
+
+      The value type of this field is STRING.
+
+    - **`compatibility score field`**
+
+      This OPTIONAL property describes the score of the compatibility set.
+
+      There is only one compatibility score field allowed in one compatibility set.
+
+      The highest score of the compatibility set means this compatibility set is the most optimal one.
+
+      If this field is not present, the score of this compatibility set is 0.
+
+      If compatibility sets have the same score, that means any of these compatibility sets pass the compatibility check is successful in matching the compatibility requirements.
+
+      The compatibility-score is used for selecting optimal image or optimal node in descending order for those compatibility sets pass the compatibility check.
+
+      The value type of this field is INT.
+
+    - **`annotation field`**
+
+      This OPTIONAL property is a field to describe human-readable information regarding to the compatibility set.
+
+      The value type of this field is STRING.
+
+### Artifact example of use
+
+All fields in this section are only for examples.
+
+#### Example - image is compatible with different hardware combinations
+
+```json
+{
+  "compatibilities": [
+    {
+      "oci.cpu.vendor": "GenuineIntel",
+      "oci.cpu.features": "AVX512FP16,",
+      "oci.kernel.selinux": "false",
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
```

Comment:
We will probably need to discuss how to handle lists (and other non-string types). Arguably you could also do an index added to the key `name.0`

Reply:
This has been discussed here https://github.com/opencontainers/wg-image-compatibility/pull/17#discussion_r1546173866, it's about setting a range for kernel versions.

When it comes to list I think we could just separate elements over white space character.

Reply:
for the value expression(lists and other non-string types etc), it could be discussed in label defintion phase.

Reply:
Agree on this point!